pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -59,14 +99,29 @@ t_ndarray   array_create(int32_t nd, int64_t *shape,
     }
     arr.buffer_size = arr.length * arr.type_size;
     arr.strides = malloc(nd * sizeof(int64_t));
-    for (int32_t i = 0; i < arr.nd; i++)
+    if (arr.order == order_c)
+    {
+        for (int32_t i = 0; i < arr.nd; i++)
+        {
+            arr.strides[i] = 1;
+            for (int32_t j = i + 1; j < arr.nd; j++)
+                arr.strides[i] *= arr.shape[j];
+        }
+    }
+    else if (arr.order == order_f)
     {
-        arr.strides[i] = 1;
-        for (int32_t j = i + 1; j < arr.nd; j++)
-            arr.strides[i] *= arr.shape[j];
+        for (int32_t i = 0; i < arr.nd; i++)
+        {
+            arr.strides[i] = 1;
+            for (int32_t j = 0; j < i; j++)
+                arr.strides[i] *= arr.shape[j];
+        }
     }
     if (!is_view)
+    {
         arr.raw_data = malloc(arr.buffer_size);
+        memset(arr.raw_data, 0, arr.buffer_size);

```

Comment:
Why did you add this `memset`? Is this for debugging purposes? You have effectively changed this function from `np.empty` to `np.zeros`

Reply:
My bad, it was for debugging purposes.